spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java

```diff
@@ -81,32 +86,29 @@
 import org.apache.spark.sql.internal.SQLConf;
 import org.apache.spark.sql.types.StructType;
 import org.jetbrains.annotations.NotNull;
-import org.junit.After;
-import org.junit.AfterClass;
-import org.junit.Assert;
-import org.junit.Assume;
-import org.junit.BeforeClass;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-@RunWith(Parameterized.class)
-public class TestSparkReaderDeletes extends DeleteReadTests {
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ParameterizedTestExtension.class)
+public class TestSparkReaderDeletes extends ParameterizedDeleteReadTest {
```

Comment:
`DeleteReadTests` and its subclasses should probably be handled in a separate PR

Reply:
As mentioned in https://github.com/apache/iceberg/pull/9380#discussion_r1436475714, I've opened a draft PR https://github.com/apache/iceberg/pull/9382 for exactly that. 

There's a blocking error in that PR I need help with. See https://github.com/apache/iceberg/pull/9382#issuecomment-1883659384.